aiida_vasp/workflows/base.py

```diff
@@ -81,8 +81,17 @@ def define(cls, spec):
         spec.output('output_kpoints', valid_type=get_data_class('array.kpoints'), required=False)
 
     def prepare_calculation(self):
+        """Set the restart folder and set INCAR tags for a restart."""
         if isinstance(self.ctx.restart_calc, self._calculation_class):
             self.ctx.inputs.restart_folder = self.ctx.restart_calc.out.remote_folder
+            old_incar = AttributeDict(self.ctx.inputs.incar.get_dict())
+            incar = old_incar.copy()
+            if 'istart' in incar:
```

Comment:
Any required changes can and should be implemented in error handlers, this is only the default behaviour if error handlers don't do anything else.

Reply:
Convergence tests would be on a higher level. On this level we only try our best to finish the calculation with inputs as close to the original ones as possible.

By error handlers I mean functions like the ones in the QE plugin's PhBaseWorkchain. They get called by the workchain to modify the inputs in order to avoid running into the same problem in the next iteration. An error handler could determine that WAVECAR should not be used and switch to only using CHGCAR etc. The restart would then be run that way, without VASP having to do anything.

Reply:
A convergence run would realistically have to perform a relaxation as well.